rfd/0067-database-access-aws-redis

```diff
@@ -0,0 +1,276 @@
+database users not managed by Teleport, users can manually input `auth`.
+
+### Secret store
+
+#### Secret store - Interface
+
+For ElastiCache/MemoryDB Redis usage, it is natural to use an AWS native
+solution (aka [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/)).
+Though other secret stores could be supported in the future. <sup>[2]<sup>
+
+Here is a sample interface for managing secrets using the secret store.
+```
+// SecretValue is the secret value.
+type SecretValue struct {
+    // Key is the key path of the secret.
+    Key string
+    // Value is the value of the secret.
+    Value string
+    // Version is the version of the secret value.
+    Version string
+    // Created is the creation time of this version.
+    Created time.Time
+}
+
+// PutValueRequest defines the request to create a new secret version.
+type PutValueRequest struct {
+    // Key is the key path of the secret.
+    Key string
+    // Value is the value of the secret to set. A random value is generated if
+    // empty.
+    Value string
+    // RandomValueLength is the length of the random value to generate.
+    RandomValueLength int
+    // CurrentVersion is the version of the current secret value. If set, it
+    // will be used for test-and-set to ensure only one caller succeeds for the
+    // same version.
+    CurrentVersion string
+}
+
+// Secrets defines an interface for managing secrets. A secret consists of a
+// key path and a list of versions that hold copies of current or past secret
+// values.
+type Secrets interface {
+    // PutValue creates a new secret version for the secret. The secret is also
+    // created if it does not exist. CurrentVersion can be provided to perform
+    // a test-and-set operation, and an error will be returned if the test
+    // fails.
+    PutValue(req PutValueRequest) (*SecretValue, error)
+
+    // GetValue returns the secret value for provided version. Besides version
+    // string returned from PutValue, two specials versions "CURRENT" and
+    // "PREVIOUS" can also be used to retrieve the current and previous
+    // versions respectively. If the version is empty, "CURRENT" is used.
+    GetValue(key string, version string) (*SecretValue, error)
+
+    // Delete deletes the secret for the provided path.
+    Delete(key string) error
+}
+```
+It is up to the caller to call `PutValue` periodically to rotate the secret.
+
+#### Secret store - AWS Secrets Manager
+
+AWS Secrets Manager is used to store passwords for ElastiCache and MemoryDB
+users. An optional `secret_store` block can be specified per database to
+overwrite default settings.
+```
+db_service:
+  enabled: "yes"
+
+  databases:
+  - name: "elasticache-example"
+    protocol: "redis"
+    uri: "master.example.xxxxxx.use1.cache.amazonaws.com"
+    aws:
+      region: "us-east-1"
+
+      # Optional secret store settings.
+      secret_store:
+        # Secert name prefix. Defaults to "teleport/".
```

Comment:
```suggestion
        # Secret name prefix. Defaults to "teleport/".
```

Also, the default value should probably include the database name in it somehow?

Reply:
ElastiCache user and user groups can be shared across ElastiCache clusters. The secret path for a particular user can be `<prefix>/elasticache/<user-id>`. Will add in the doc.